### ISSUE TITLE
Unsupported `sprintf` for Arduino Uno

### ISSUE THREAD

**Issue by almonds0166** [almonds0166/6302view#4]
I've noticed last semester and just today that my Arduino Uno scripts haven't been building the strings correctly (on both Linux Mint and Windows). I'm not sure whether support for `sprintf` has been dropped in only the last couple of years or whether this been a problem since the beginning of 6302view and I just haven't noticed (how??).

I think I'll avoid byte-heavy [streams](https://playground.arduino.cc/Main/Printf/) and go a lower-level route, sifting through the cpp and taking out all the `%.*s` format specifiers and otherwise broken `sprintf` calls and replace them with `dtostrf`, `strncpy`, and `strcat` calls.

I'll also play around with the CSS.

Here's an amusing example script I'll use for testing:

```cpp
#include <Six302.h>

// microseconds
#define STEP_TIME 10000
#define REPORT_TIME 50000

CommManager cm(STEP_TIME, REPORT_TIME);

bool toggle = false;
bool button;
float slider;

float plot;
int32_t number = 123;

void setup() {
  /* Add modules */
  cm.addToggle(&toggle, "(Toggle) really really really long title");
  cm.addButton(&button, "(Button) really really really long title");
  cm.addSlider(&slider, "(Slider) really really really long title", -3, 3, 0.1);
  
  cm.addPlot(&plot, "(Plot) really really really long title", -10, 10);
  cm.addNumber(&number, "(Number) really really really long title");
  
  /* Ready to communicate over Serial */
  cm.connect(&Serial, 115200);
}

void loop() {
  plot = slider * slider;
  cm.step();
}
```

**Comment by almonds0166** [comment]
Still on my mind, just too busy

**Comment by almonds0166** [comment]
~~This issue has been bugging me at least once a week for the last 8 months, but~~ I finally fixed this issue! (:

Interestingly, since we've replaced all the `sprintf` calls in favor of lower level options on the Arduino Uno, we've gained a bit of progmem. I was worried the opposite would happen, but in retrospect, this makes sense. Looking through the code, I see further opportunities for improvement, but it's not a priority.

:tada:
